src/shared/telemetry/telemetryEvent.ts

```diff
@@ -41,7 +42,12 @@ export function toMetricData(array: TelemetryEvent[]): MetricDatum[] {
                     }
 
                     return {
-                        MetricName: `${metricEvent.namespace}_${datum.name}`.replace(NAME_ILLEGAL_CHARS_REGEX, ''),
+                        // MetricName is the namespace without underscores and the name without underscores
+                        // concatenated with an underscore and passed through the service definition's regex
+                        // (alphanumeric + underscores + `+-.` + :)
+                        // tslint:disable-next-line:max-line-length
+                        MetricName: `${metricEvent.namespace.replace(REMOVE_UNDERSCORES_REGEX, '')}_${datum.name.replace(REMOVE_UNDERSCORES_REGEX, '')}`
+                            .replace(NAME_ILLEGAL_CHARS_REGEX, ''),
```

Comment:
Consider something like this so the code reads nicer:

```
                        MetricName: `${sanitizeX(metricEvent.namespace)}_${sanitizeX(datum.name)}`
                            .replace(NAME_ILLEGAL_CHARS_REGEX, ''),
```

Reply:
re: this and the past conversation, do we want to enforce a double-regex replace like this or should we just mark a spot on our coding standards that says that we shouldn't include underscores in the metric names? 

Regardless, we shouldn't have to enforce this for namespaces since those are enums.